doc/modules/model_evaluation.rst

```diff
@@ -1109,6 +1110,71 @@ function:
     >>> matthews_corrcoef(y_true, y_pred)  # doctest: +ELLIPSIS
     -0.33...
 
+.. _multilabel_confusion_matrix:
+
+Multi-label confusion matrix
+----------------
+
+The :func:`multilabel_confusion_matrix` function computes multilabel confusion
+matrix class-wisely (default) or sample-wisely (samplewise=True) to evaluate
+the accuracy of a classification. multilabel_confusion_matrix also treats
+multiclass data as if it were multilabel, as this is a transformation commonly
+applied to evaluate multiclass problems with binary classification metrics
+(such as precision, recall, etc.).
+
+When calculating class-wise multilabel confusion matrix :math:`MCM`, the
+count of true negatives for class :math:`i` is :math:`MCM_{i,0,0}`, false
+negatives is :math:`MCM_{i,1,0}`, true positives is :math:`MCM_{i,1,1}`
+and false positives is :math:`MCM_{i,0,1}`.
+
+Here is an example demonstrating the use of the
+:func:`multilabel_confusion_matrix` function in a multilabel_indicator
+problem::
+
+    >>> from sklearn.metrics import multilabel_confusion_matrix
+    >>> y_true = [[1, 0, 1],
+    ...           [0, 1, 0]]
+    >>> y_pred = [[1, 0, 0],
+    ...           [0, 1, 1]]
+    >>> multilabel_confusion_matrix(y_true, y_pred)
+    array([[[1, 0],
+            [0, 1]],
+    <BLANKLINE>
+           [[1, 0],
+            [0, 1]],
+    <BLANKLINE>
+           [[0, 1],
+            [1, 0]]])
+
+Here is an example demonstrating the use of the
+:func:`multilabel_confusion_matrix` function in a multiclass problem::
+
+    >>> y_true = ["cat", "ant", "cat", "cat", "ant", "bird"]
+    >>> y_pred = ["ant", "ant", "cat", "cat", "ant", "cat"]
+    >>> multilabel_confusion_matrix(y_true, y_pred,
+    ...                             labels=["ant", "bird", "cat"])
+    array([[[3, 1],
+            [0, 2]],
+    <BLANKLINE>
+           [[5, 0],
+            [1, 0]],
+    <BLANKLINE>
+           [[2, 1],
+            [1, 2]]])
+
+Here is an example demonstrating the use of the
```

Comment:
I think this can be removed. binary here is just a degenerate multiclass case, not of great interest to users